src/main/java/org/alfresco/rest/api/impl/DeletedNodesImpl.java

```diff
@@ -189,4 +200,35 @@ public void purgeArchivedNode(String archivedId)
         NodeRef validatedNodeRef = nodes.validateNode(StoreRef.STORE_REF_ARCHIVE_SPACESSTORE, archivedId);
         nodeArchiveService.purgeArchivedNode(validatedNodeRef);
     }
+
+    @Override
+    public BinaryResource readProperty(String archivedId, String renditionId, Parameters parameters)
+    {
+        // First check if the archived node is valid
+        NodeRef validatedNodeRef = nodes.validateNode(StoreRef.STORE_REF_ARCHIVE_SPACESSTORE, archivedId);
+
+        if (renditionId != null)
```

Comment:
I'm not sure having this method do two different things is warranted - getting a rendition's content or getting a file's content. My worry would be that this ends up with all sorts of conditional logic that could lead to weird bugs over time, whereas two separate methods would allow the removal of the conditional logic entirely.

It would be easy to merge them later if there is some reason they should be tied together.